src/NuGet.Core/NuGet.Commands/NuGet.Commands.csproj

```diff
@@ -64,5 +64,6 @@
     <InternalsVisibleTo Include="NuGet.ProjectModel.Test" />
     <InternalsVisibleTo Include="NuGet.SolutionRestoreManager.Test" />
     <InternalsVisibleTo Include="Test.Utility" />
+    <InternalsVisibleTo Include="Dotnet.Integration.Test" />
```

Comment:
dotnet.integration.test writes project files to disk, executes the dotnet CLI as a child process, then reads files off disk to validate.  There's no reason why it should need access to internal classes or methods.

Reply:
That is for testing purposes. To allow access to the resex file which is internal

Reply:
Can we assert the error code only in those tests? 

That can be done in a follow-up ofc.

Reply:
After writing this, I see that we have sources tests which don't really have error codes.